src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -964,13 +952,250 @@ def load_module(name, value):
                 f"Pipeline {pipeline_class} expected {expected_modules}, but only {passed_modules} were passed."
             )
 
-        # 5. Instantiate the pipeline
+        # 8. Instantiate the pipeline
         model = pipeline_class(**init_kwargs)
 
+        return_cached_folder = kwargs.pop("return_cached_folder", False)
         if return_cached_folder:
+            message = f"Passing `return_cached_folder=True` is deprecated and will be removed in `diffusers=0.17.0`. Please do the following instead: \n 1. Load the cached_folder via `cached_folder={cls}.download({pretrained_model_name_or_path})`. \n 2. Load the pipeline by loading from the cached folder: `pipeline={cls}.from_pretrained(cached_folder)`."
```

Comment:
Just a question, why is this not desired anymore?

Reply:
Because if one wants the cached_folder one can just split:

```py
from_pretrained(....)
```

into

```py
cached_folder = download(...)
from_pretrained(cached_folder)
```